scionlab/models/core.py

```diff
@@ -1131,7 +1131,15 @@ def __str__(self):
 
     def _br_idx(self):
         # slow!
-        return self.AS.border_routers.filter(pk__lt=self.pk).count() + 1
+        try:
+            return self.AS.border_routers.filter(pk__lt=self.pk).count() + 1
+        except ValueError:
+            # When deleting a router we do not have it available anymore, but we are still
+            # trying to calculate its ID. As we are not showing it on the UI, it's safe
+            # to assume an arbitrary ID here.
+            return 0
```

Comment:
This works, but I don't like raising the exception in the first place. A slightly more straightforward way could be:
```python
if self.pk is None:
  return 0
return ... query ...
```
Checking for an existing `self.pk` seems a fairly common way to check that the object is currently in the DB. Interpreting the stack trace in the related issue, this should work here too.

Reply:
Indeed looks more elegant